parsing/lexer.mll

```diff
@@ -294,7 +322,14 @@ let lax_delim raw_name =
      if Utf8_lexeme.is_lowercase name then Some name
      else None
 
-let is_keyword name = Hashtbl.mem keyword_table name
+let is_keyword name =
+  Hashtbl.mem keyword_table name
+
+let find_keyword lexbuf name =
+  match Hashtbl.find keyword_table name with
+  | Some x -> x
+  | None -> error lexbuf (Unknown_keyword name)
+  | exception Not_found -> LIDENT name
```

Comment:
This might be clearer with a custom type:

```ocaml
let find_keyword lexbuf name =
  match Hashtbl.find keyword_table name with
  | Known x -> x
  | Future -> error lexbuf (Unknown_keyword name)
  | exception Not_found -> LIDENT name
```